std/conv.d

```diff
@@ -2870,6 +2907,15 @@ Target parse(Target, Source)(ref Source p)
     return (sign) ? -ldval : ldval;
 }
 
+///
+@safe unittest
+{
+    import std.math : approxEqual;
+    auto str = "123.456";
+
+    assert(parse!double(str).approxEqual(123.456));
```

Comment:
You have a point, but the vagaries of the precision of intermediate values may change things slightly.